actors/common/src/main/java/org/sunbird/learner/actors/skill/UserSkillManagementActor.java

```diff
@@ -3,68 +3,186 @@
-import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.models.util.*;
 import org.sunbird.common.models.util.ProjectUtil.EsType;
-import org.sunbird.common.models.util.TelemetryEnvKey;
 import org.sunbird.common.models.util.datasecurity.OneWayHashing;
 import org.sunbird.common.request.ExecutionContext;
 import org.sunbird.common.request.Request;
 import org.sunbird.common.responsecode.ResponseCode;
 import org.sunbird.dto.SearchDTO;
 import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.actors.skill.dao.UserSkillDao;
+import org.sunbird.learner.actors.skill.dao.impl.UserSkillDaoImpl;
 import org.sunbird.learner.util.Util;
+import org.sunbird.models.user.skill.Skill;
 import org.sunbird.telemetry.util.TelemetryUtil;
 
 /**
  * Class to provide functionality for Add and Endorse the user skills . Created by arvind on
  * 18/10/17.
  */
 @ActorConfig(
-  tasks = {"addSkill", "getSkill", "getSkillsList"},
+  tasks = {"addSkill", "getSkill", "getSkillsList", "updateSkill"},
   asyncTasks = {}
 )
-public class SkillmanagementActor extends BaseActor {
+public class UserSkillManagementActor extends BaseActor {
 
   private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
   private Util.DbInfo userSkillDbInfo = Util.dbInfoMap.get(JsonKey.USER_SKILL_DB);
   private Util.DbInfo skillsListDbInfo = Util.dbInfoMap.get(JsonKey.SKILLS_LIST_DB);
   private Util.DbInfo userDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
   private static final String REF_SKILLS_DB_ID = "001";
+  private UserSkillDao userSkillDao = UserSkillDaoImpl.getInstance();
 
   @Override
   public void onReceive(Request request) throws Throwable {
-
+    String operation = request.getOperation();
     Util.initializeContext(request, TelemetryEnvKey.USER);
     // set request id fto thread loacl...
     ExecutionContext.setRequestId(request.getRequestId());
-    if (request.getOperation().equalsIgnoreCase(ActorOperations.ADD_SKILL.getValue())) {
-      endorseSkill(request);
-    } else if (request.getOperation().equalsIgnoreCase(ActorOperations.GET_SKILL.getValue())) {
-      getSkill(request);
-    } else if (request
-        .getOperation()
-        .equalsIgnoreCase(ActorOperations.GET_SKILLS_LIST.getValue())) {
-      getSkillsList();
-    } else {
-      onReceiveUnsupportedOperation(request.getOperation());
+
+    switch (operation) {
+      case "addSkill":
+        endorseSkill(request);
+        break;
+      case "getSkill":
+        getSkill(request);
+        break;
+      case "getSkillsList":
+        getSkillsList();
+        break;
+      case "updateSkill":
+        updateSkill(request);
+        break;
+      default:
+        onReceiveUnsupportedOperation("BadgeClassActor");
+    }
+  }
+  /** Method will Update the skills from user ... */
+  private void updateSkill(Request actorMessage) {
+    ProjectLogger.log(
+        "SkillmanagementActor-updateSkill called",
+        actorMessage.getRequest(),
```

Comment:
Is the request being logged or the object reference? In case object reference is logged, it would not be of much use.